### PR TITLE
PIMS-422 Projects stubs

### PR DESCRIPTION
**Pull request by GrahamS-Quartech**
## 🎯 Summary

[PIMS-422: ](https://apps.itsm.gov.bc.ca/jira/browse/PIMS-422)

* Added stubs for routes, controllers, and tests.
* Added some fairly robust factories in the tests but these objects are kind of enormous so I simplified slightly in some spots.
* I skipped routes that had /report as their root cause I think that belongs in the reports ticket. There were also some routes already implemented by the notifications ticket.
* Left some notes inline for the routes again, but these ones mostly make sense. Will add to the issue discussion as well.

## 🔰 Checklist

- [x] I have read and agree with the following checklist and am following the guidelines in our [Code of Conduct](CODE_OF_CONDUCT.md) document.

> - I have performed a self-review of my code.
> - I have commented my code, particularly in hard-to-understand areas.
> - I have made corresponding changes to the documentation where required.
> - I have tested my changes to the best of my ability.
> - My changes generate no new warnings.

### CODE REVIEW

File: express-api/src/controllers/projects/projectsController.ts [hunk 1]
```diff
@@ -0,0 +1,325 @@
...
+   */
+  return stubResponse(res);
+};
+
+/**
+ * @description Update the report for the specified project.
+ * @param {Request}     req Incoming request.
+ * @param {Response}    res Outgoing response.
+ * @returns {Response}      A 200 status with the deleted report.
+ */
```
Comment (dbarkowsky) [inline express-api/src/controllers/projects/projectsController.ts:128]:
```suggestion
 * @description Delete the report for the specified project.
 * @param {Request}     req Incoming request.
 * @param {Response}    res Outgoing response.
 * @returns {Response}      A 200 status with the deleted report.
 */
```
I wonder if we should also consider what kind of response we want to return for deletions. 200 makes sense I guess if we're returning something, but I feel like I've seen a lot of cases in the original that could have just sent a 204 response back.

File: express-api/src/routes/projectRouter.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+const router = express.Router();
+
+export const PROJECT_DISPOSAL = '/projects/disposal';
+export const PROJECT_REPORTS = '/projects/reports';
+
+//These originally had a separate route for numeric id and projectNumber, but I don't think express supports this pattern.
+router.route(`${PROJECT_DISPOSAL}/:projectId`).get(controllers.getDisposalProject);
+router.route(`${PROJECT_DISPOSAL}/:projectId`).put(controllers.updateDisposalProject);
+router.route(`${PROJECT_DISPOSAL}/:projectId`).delete(controllers.deleteDisposalProject);
```
Comment (dbarkowsky) [inline express-api/src/routes/projectRouter.ts:12]:
```suggestion
router.route(`${PROJECT_DISPOSAL}/:projectId`)
  .get(controllers.getDisposalProject)
  .put(controllers.updateDisposalProject)
  .delete(controllers.deleteDisposalProject);
```
If the route is the same, you can always chain the methods together like this. Might be easier to read.

File: express-api/src/routes/projectRouter.ts [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+router.route(`${PROJECT_DISPOSAL}`).post(controllers.addDisposalProject);
+
+//This originally had like 3 different routes that all seem to do the same thing so let's just consolidate that to one.
+router.route(`${PROJECT_DISPOSAL}/workflows`).put(controllers.requestProjectStatusChange);
+
+router.route(`${PROJECT_REPORTS}`).get(controllers.getAllProjectReports);
+router.route(`${PROJECT_REPORTS}/:reportId`).get(controllers.getProjectReport);
+router.route(`${PROJECT_REPORTS}/:reportId`).put(controllers.updateProjectReport);
+router.route(`${PROJECT_REPORTS}/:reportId`).delete(controllers.deleteProjectReport);
```
Comment (dbarkowsky) [inline express-api/src/routes/projectRouter.ts:21]: Same comment as above.

File: express-api/src/controllers/projects/projectsController.ts (hunk 1 above)
Comment (GrahamS-Quartech) [reply express-api/src/controllers/projects/projectsController.ts:128]: Either way would definitely be fine, I guess it just depends on whether the frontend might be able to do something useful with the entire deleted resource or not (ie. show the user that the snapshots were deleted in addition to the requested report)

**Review by dbarkowsky — APPROVED** [review]

**Pull request merged**